### PR TITLE
Clean session modification & sync nng submodule

### PR DESCRIPTION
**Pull request by wanghaEMQ**

### CODE REVIEW

**Review by lee-emqx — APPROVED** [review]

**Review by lee-emqx — APPROVED** [review]

File: nanomq/apps/broker.c [hunk 1]
```diff
@@ -227,15 +227,8 @@ server_cb(void *arg)
 				// avoid being free
 				conn_param_clone(work->cparam);
 			}
-			// restore clean session
-			char *clientid =
-			    (char *) conn_param_get_clientid(work->cparam);
-			if (clientid != NULL) {
-				restore_session(clientid, work->cparam,
```
Comment (alvin1221) [inline nanomq/apps/broker.c:234]: Why delete `restore_session`?

**Review by alvin1221 — APPROVED** [review]

File: nanomq/apps/broker.c (hunk 1 above)
Comment (wanghaEMQ) [reply nanomq/apps/broker.c:234]: Clean session functions would be moved to the protocol layer.

File: nanomq/include/sub_handler.h [hunk 2]
```diff
@@ -17,7 +17,7 @@ void destroy_sub_pkt(packet_subscribe *, uint8_t);
 void init_sub_property(packet_subscribe *);
 
 // functions about clean session
-int  cache_session(char *, conn_param *, uint32_t, void *);
-int  restore_session(char *, conn_param *, uint32_t, void *);
+int  cache_session(uint32_t, conn_param *, uint32_t, void *);
+int  restore_session(uint32_t, conn_param *, uint32_t, void *);
```
Comment (JaylinYu) [inline nanomq/include/sub_handler.h:21]: seems like nowhere is using it?

**Review by JaylinYu — APPROVED** [review]

File: nanomq/include/sub_handler.h (hunk 2 above)
Comment (wanghaEMQ) [reply nanomq/include/sub_handler.h:21]: yes, I will remove it later

**Review by wanghaEMQ — COMMENTED** [review]
It's okay.

**Pull request merged**